### PR TITLE
[fastlane_core] Add an `os_type` property to the `Device` class

### PR DESCRIPTION
**Pull request by 0xced**
- [x] Run `rspec` for all tools you modified
- [x] Run `rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

---

This will be required for an upcoming change to `scan` for handling projects supporting multiple platforms with a single scheme as described in http://promisekit.org/news/2016/08/Multiplatform-Single-Scheme-Xcode-Projects/

### CODE REVIEW

**Review by KrauseFx — APPROVED** [review]
This looks good to me, great addition to _fastlane_core_. I'd love to have a second review before we merge this 👍

**Comment by 0xced** [comment]
I was very impressed that adding a new parameter in the middle of the `Device` constructor did not break dependencies (such as scan) which don't know about that parameter!

Or I did not actually test scan with my local copy of fastlane_core. 😄

**Comment by 0xced** [comment]
I just rebased, I hope it can be part of fastlane_core 0.52.4. ☺️

**Review by asfalcone — APPROVED** [review]

**Comment by mfurtak** [comment]
Do I understand correctly that this parameter doesn't actually reflect the OS of the devices we find, but rather, the OS of the devices we ask for? Meaning, if we don't specify a `requested_os_type`, then all `Device`s come back with a `os_type` of `''`?

**Comment by 0xced** [comment]
Actually, the `connected_devices(requested_os_type)` function returns early if `requested_os_type` is not `iOS` or `tvOS`, so the `os_type` attribute will always be valid.

**Comment by mfurtak** [comment]
Cool. Is the same true for `simulators`?

**Comment by 0xced** [comment]
For the simulators, `os_type` (the one parsed from `xcrun simctl list`) is used instead of `requested_os_type`, so it will also be always valid.

**Comment by mfurtak** [comment]
Just got around to pulling the branch and giving it a try. It seems great - nice job! 👍

**Pull request merged**
